### PR TITLE
[CWS] Properly set process exec child for runtime events

### PR DESCRIPTION
**Pull request by lebauce**
### What does this PR do?

Set IsExecChild according to parent's IsThread attribute

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/security/security_profile/activity_tree/activity_tree.go [hunk 1]
```diff
@@ -452,6 +452,8 @@ func (at *ActivityTree) CreateProcessNode(entry *model.ProcessCacheEntry, branch
 		// we haven't found anything, create a new ProcessActivityNode for the input processCacheEntry
 		if !dryRun {
 			node = NewProcessNode(entry, generationType, resolvers)
+			node.Process.IsExecChild = entry.Parent != nil && entry.Ancestor.Parent != nil && entry.Ancestor.Parent.Pid == entry.Process.Pid
```
Comment (safchain) [inline pkg/security/security_profile/activity_tree/activity_tree.go:455]: Is this still required ?

File: pkg/security/secl/model/process_cache_entry.go [hunk 2]
```diff
@@ -78,7 +78,7 @@ func (pc *ProcessCacheEntry) Exec(entry *ProcessCacheEntry) {
 
 	// use exec time as exit time
 	pc.Exit(entry.ExecTime)
-	entry.Process.IsExecChild = true
+	entry.Process.IsExecChild = !pc.IsThread
```
Comment (safchain) [inline pkg/security/secl/model/process_cache_entry.go:81]: IsExecExec

File: docs/cloud-workload-security/backend.schema.json [hunk 3]
```diff
@@ -863,9 +863,9 @@
           "type": "boolean",
           "description": "Indicates whether the process is a kworker"
         },
-        "is_exec_child": {
+        "is_exec_exec": {
           "type": "boolean",
-          "description": "Indicates wether the process is an exec child of its parent"
+          "description": "Indicates wether the process is an exec following an other exec"
```
Comment (rtrieu) [inline docs/cloud-workload-security/backend.schema.json:868]:
```suggestion
          "description": "Indicates whether the process is an exec following another exec"
```

File: docs/cloud-workload-security/backend.schema.json [hunk 4]
```diff
@@ -987,9 +987,9 @@
...
```
Comment (rtrieu) [inline docs/cloud-workload-security/backend.schema.json:992]:
```suggestion
          "description": "Indicates whether the process is an exec following another exec"
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Some small change requests, but otherwise looks good.

**Review by rtrieu — CHANGES_REQUESTED** [review]
Hi, still seeing one small typo

File: docs/cloud-workload-security/backend.schema.json [hunk 5]
```diff
@@ -863,9 +863,9 @@
           "type": "boolean",
           "description": "Indicates whether the process is a kworker"
         },
-        "is_exec_child": {
+        "is_exec_exec": {
           "type": "boolean",
-          "description": "Indicates wether the process is an exec child of its parent"
+          "description": "Indicates wether the process is an exec following another exec"
```
Comment (rtrieu) [inline docs/cloud-workload-security/backend.schema.json:868]:
```suggestion
          "description": "Indicates whether the process is an exec following another exec"
```

File: docs/cloud-workload-security/backend.schema.json [hunk 6]
```diff
@@ -987,9 +987,9 @@
...
```
Comment (rtrieu) [inline docs/cloud-workload-security/backend.schema.json:992]:
```suggestion
          "description": "Indicates whether the process is an exec following another exec"
```

File: pkg/security/secl/model/model.go [hunk 7]
```diff
@@ -656,8 +656,8 @@ type Process struct {
 	ScrubbedArgsTruncated bool           `field:"-" json:"-"`
 	Variables             eval.Variables `field:"-" json:"-"`
 
-	IsThread    bool `field:"is_thread"` // SECLDoc[is_thread] Definition:`Indicates whether the process is considered a thread (that is, a child process that hasn't executed another program)`
-	IsExecChild bool `field:"-"`         // Indicates whether the process is an exec child of its parent
+	IsThread   bool `field:"is_thread"` // SECLDoc[is_thread] Definition:`Indicates whether the process is considered a thread (that is, a child process that hasn't executed another program)`
+	IsExecExec bool `field:"-"`         // Indicates wether the process is an exec following another exec
```
Comment (rtrieu) [inline pkg/security/secl/model/model.go:660]:
```suggestion
	IsExecExec bool `field:"-"`         // Indicates whether the process is an exec following another exec
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Hey sorry not sure if I missed them before, but seeing typos of the same word.

**Review by spikat — APPROVED** [review]

**Review by rtrieu — APPROVED** [review]
Looks good!

**Pull request merged**